Makefile.defs

```diff
@@ -228,23 +239,26 @@ else ifeq ($(TARGET), ubuntu-touch)
 	ARM_ARCH+=-mfloat-abi=hard
 	COMPAT_CFLAGS:=$(UBUNTU_COMPAT_CFLAGS)
 	COMPAT_CXXFLAGS:=$(UBUNTU_COMPAT_CFLAGS)
-else ifeq ($(TARGET), kindle)
-	ARM_ARCH:=$(ARMV7_A8_ARCH)
-	ARM_ARCH+=-mfloat-abi=softfp
-	COMPAT_CFLAGS:=$(UBUNTU_COMPAT_CFLAGS)
-	COMPAT_CXXFLAGS:=$(UBUNTU_COMPAT_CFLAGS)
 else ifeq ($(TARGET), pocketbook)
 	ARM_ARCH:=$(ARMV7_A8_ARCH)
 	ARM_ARCH+=-mfloat-abi=softfp
 	COMPAT_CFLAGS:=$(MG2K12_COMPAT_CFLAGS) -std=gnu99 --sysroot $(SYSROOT)
 	COMPAT_CXXFLAGS:=$(MG2K12_COMPAT_CXXFLAGS) --sysroot $(SYSROOT)
-else ifeq ($(TARGET), kindle5)
+else ifeq ($(TARGET), kindle)
 	ARM_ARCH:=$(ARMV7_A8_ARCH)
 	ARM_ARCH+=-mfloat-abi=softfp
+	MACHINE_KINDLE:=$(shell PATH='$(PATH)' $(CC) -dumpmachine 2>/dev/null)
+    ifeq ($MACHINE_KINDLE, arm-linux-gnueabi)
+        COMPAT_CFLAGS:=$(UBUNTU_COMPAT_CFLAGS)
+        COMPAT_CXXFLAGS:=$(UBUNTU_COMPAT_CFLAGS)
+    endif
 else ifeq ($(TARGET), kindlepw2)
 	ARM_ARCH:=$(ARMV7_A9_ARCH)
 	ARM_ARCH+=-mfloat-abi=softfp
-else ifeq ($(TARGET), kindle-legacy)
+    ifeq ($MACHINE_KINDLE, arm-linux-gnueabi)
```

Comment:
Ah, I see where you got the PATH thing from ;).

Another solution is to move that MACHINE assignment up, before the TARGET dance (somewhere after we set $(CC)), and reuse it in your tests ;).

Reply:
Oh whoops, you're right. It's mysterious-ish but not *that* mysterious. :-P

Reply:
And in that case, I'd *definitely* want to assign the MACHINE var with `:=`, we do *NOT* want a mismatch between the actual -dumpmachine of the compiler we'll use and a stray MACHINE var in our env.

Reply:
I'm done for that round of comments ;).

Reply:
(I have no strong preference on which way to go, but since we already re-use MACHINE in multiple places, it probably makes a tiny bit more sense to do the same).